lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -1,50 +1,124 @@
 module Datadog
   # RailsRendererPatcher contains function to patch Rails rendering libraries.
+  # rubocop:disable Lint/RescueException
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/BlockLength
   module RailsRendererPatcher
     module_function
 
     def patch_renderer
-      patch_renderer_render_template
-      patch_renderer_render_partial
+      if defined?(::ActionView::TemplateRenderer) && defined?(::ActionView::PartialRenderer)
+        patch_template_renderer(::ActionView::TemplateRenderer)
+        patch_partial_renderer(::ActionView::PartialRenderer)
+      elsif defined?(::ActionView::Rendering) && defined?(::ActionView::Partials::PartialRenderer)
+        # NOTE: Rails < 3.1 compatibility: different classes are used
```

Comment:
the same class patcher is fine for all Rails versions (>= 3.2) while we need to handle some little differences in Rails < 3.2